includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -294,37 +292,27 @@ public function filter_gateway_title( $title, $id ) {
 	}
 
 	/**
-	 * Removes postal code validation from WC.
+	 * Normalizes postal code in case of redacted data from Apple Pay.
 	 *
-	 * @param bool   $valid Whether postal code is valid.
 	 * @param string $postcode Postal code.
 	 * @param string $country Country.
-	 * @return bool Whether postal code is valid.
 	 */
-	public function postal_code_validation( $valid, $postcode, $country ) {
-		$gateways = WC()->payment_gateways->get_available_payment_gateways();
-
-		if ( ! isset( $gateways['woocommerce_payments'] ) ) {
-			return $valid;
-		}
-
-		$payment_request_type = isset( $_POST['payment_request_type'] ) ? wc_clean( wp_unslash( $_POST['payment_request_type'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification
-
-		if ( 'apple_pay' !== $payment_request_type ) {
-			return $valid;
-		}
-
+	public function get_normalized_postal_code( $postcode, $country ) {
 		/**
-		 * Currently Apple Pay truncates postal codes from UK and Canada to first 3 characters
+		 * Currently Apple Pay truncates postal codes from UK and Canada to first 3-4 characters
```

Comment:
It's not clear from the comment how many characters for each country, it's actually 4 for the UK and 3 for Canada

```suggestion
		 * Currently, Apple Pay truncates the UK and Canadian postal codes to the first 4 and 3 characters respectively
```